File: Ileya/Day 1/Python/Level 2/task_2.py
def process_duplicates(arr):
    result = []
    range_ = len(arr)
    for index in range(range_):
        output = []
        duplicates = []
        is_duplicate = False
        for instance in range (index):
            if arr[index] == arr[instance]:
                if is_duplicate == True:
                    duplicates.append(index)
                    break
                is_duplicate = True
                duplicates.append(instance)
                duplicates.append(index)
                
        if is_duplicate == True:
            output.append(arr[index])
            output.append(duplicates)
            result.append(output)

    result.reverse()

    return result

File: Ileya/Day 1/JavaScript/Level 2/Task2.js
function processDuplicates(arr){
    let result = [];
    for (let index = 0; index < arr.length; index++) {
        let output = [];
        let duplicates = [];
        let isDuplicate = false;

        for (let instance = 0; instance < index; instance++) {
            if (arr[index] == arr[instance]){
                if (isDuplicate == true) {
                    duplicates.push(index)
                    break
                }

                isDuplicate = true
                duplicates.push(instance)
                duplicates.push(index)
            }
        }
                
        if (isDuplicate == true) {
            output.push(arr[index])
            output.push(duplicates)
            result.push(output)
        }
    }

    result.reverse()

    return result
}

module.exports = processDuplicates;
